### PR TITLE
Fix full page redirection at top level

### PR DESCRIPTION
**Pull request by paulomarg**
### What this PR does

When we moved to the CDN app bridge, we introduced an unexpected error: when loading at the top level, it triggered an unexpected redirect that broke authentication in some cases.

This PR checks whether the app is embedded or not before rendering the app bridge script tag, so it doesn't load in a top-level scenario.

### Reviewer's guide to testing

This only happens when the app hits the login page without `?top_level=true` while the app is uninstalled.

### Things to focus on

1. Are we skipping any edge cases?

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users

### CODE REVIEW

File: test/shopify_app/controller_concerns/login_protection_test.rb [hunk 1]
```diff
@@ -603,14 +620,25 @@ class LoginProtectionControllerTest < ActionController::TestCase
...
     assert_select "[id=redirection-target]", 1 do |elements|
       assert_equal "{\"myshopifyUrl\":\"https://#{shop_domain}\",\"url\":\"#{example_url}\"}",
         elements.first["data-target"]
     end
+
+    if expect_embedded
+      assert_select "script", 2 do |elements|
+        assert_equal "https://cdn.shopify.com/shopifycloud/app-bridge.js", elements[0]["src"]
+        assert_match %r/\/assets\/shopify_app\/redirect-[^\.]*\.js/, elements[1]["src"]
+      end
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply test/shopify_app/controller_concerns/login_protection_test.rb:636]: Sure, I can do that :)

**Review by sle-c — APPROVED** [review]
Code logic LGTM
